.editorconfig

```diff
@@ -0,0 +1,51 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2018.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# EditorConfig sets project-wide editor defaults: https://EditorConfig.org
+
+# top-most EditorConfig file, stop looking higher in the tree
+root = true
+
+# Default settings can be overidden by editorconfig file in a subdir
+# or by a specific glob later in this file
+[*]
+end_of_line = lf
+insert_final_newline = true
+charset = utf-8
+indent_style = space
+trim_trailing_whitespace = true
+
+# Python
+[*.py]
+indent_size = 4
+
+# Javascript
+[*.{js,json}]
```

Comment:
Also, I think all our `json` files are not human-generated.

Reply:
I think that these are: https://github.com/Qiskit/qiskit-terra/tree/master/qiskit/schemas (they're at least hand edited frequently) although all the schemas are deprecated and will be removed soon.

All the json files in https://github.com/Qiskit/qiskit-terra/tree/master/qiskit/test/mock/backends are machine generated and not expected to be editted

Reply:
I guess the schema files are inconsistently indented, but lean more towards 4-space than 2-space.

Reply:
Any preference for 2-vs-4 spaces for json?

Reply:
I would probably go 4 spaces just to be consistent with the python style we use.

Reply:
FWIW, since this was originally opened the schema files finished their deprecation cycle and the schema files have been removed from terra. That being said I ran `tokei` to find where we still have json files committed in the repo and there are 2 types. The first are api payload snapshots for the fake backends which are machine generated. The second are the user style guides which are hand generated.

```
========================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================
 Language                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Files        Lines         Code     Comments       Blanks
========================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================
 JSON                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           105         1121         1121            0            0
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ./qiskit/visualization/styles/iqx-dark.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 202          202            0            0
 ./qiskit/visualization/styles/default.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  202          202            0            0
 ./qiskit/visualization/styles/iqx.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      202          202            0            0
 ./qiskit/visualization/styles/bw.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       202          202            0            0
 ./qiskit/test/mock/backends/rochester/conf_rochester.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     1            1            0            0
 ./qiskit/test/mock/backends/santiago/conf_santiago.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       1            1            0            0
 ./qiskit/test/mock/backends/quito/conf_quito.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             1            1            0            0
 ./qiskit/test/mock/backends/ourense/conf_ourense.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         1            1            0            0
 ./qiskit/test/mock/backends/rome/props_rome.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              1            1            0            0
 ./qiskit/test/mock/backends/jakarta/props_jakarta.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        1            1            0            0
 ./qiskit/test/mock/backends/yorktown/props_yorktown.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      1            1            0            0
 ./qiskit/test/mock/backends/quito/props_quito.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            1            1            0            0
 ./qiskit/test/mock/backends/valencia/props_valencia.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      1            1            0            0
 ./qiskit/test/mock/backends/ourense/props_ourense.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        1            1            0            0
 ./qiskit/test/mock/backends/jakarta/conf_jakarta.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         1            1            0            0
 ./qiskit/test/mock/backends/london/props_london.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          1            1            0            0
 ./qiskit/test/mock/backends/lagos/conf_lagos.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             1            1            0            0
 ./qiskit/test/mock/backends/valencia/defs_valencia.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       1            1            0            0
 ./qiskit/test/mock/backends/johannesburg/conf_johannesburg.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                               1            1            0            0
 ./qiskit/test/mock/backends/cambridge/conf_cambridge.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     1            1            0            0
 ./qiskit/test/mock/backends/rome/defs_rome.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               1            1            0            0
 ./qiskit/test/mock/backends/santiago/defs_santiago.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       1            1            0            0
 ./qiskit/test/mock/backends/quito/defs_quito.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             1            1            0            0
 ./qiskit/test/mock/backends/burlington/conf_burlington.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   1            1            0            0
 ./qiskit/test/mock/backends/tokyo/props_tokyo.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            1            1            0            0
 ./qiskit/test/mock/backends/casablanca/conf_casablanca.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   1            1            0            0
 ./qiskit/test/mock/backends/yorktown/defs_yorktown.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       1            1            0            0
 ./qiskit/test/mock/backends/burlington/props_burlington.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  1            1            0            0
 ./qiskit/test/mock/backends/bogota/props_bogota.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          1            1            0            0
 ./qiskit/test/mock/backends/bogota/conf_bogota.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           1            1            0            0
 ./qiskit/test/mock/backends/lima/conf_lima.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               1            1            0            0
 ./qiskit/test/mock/backends/essex/conf_essex.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             1            1            0            0
 ./qiskit/test/mock/backends/athens/props_athens.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          1            1            0            0
 ./qiskit/test/mock/backends/belem/props_belem.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            1            1            0            0
 ./qiskit/test/mock/backends/boeblingen/conf_boeblingen.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   1            1            0            0
 ./qiskit/test/mock/backends/athens/conf_athens.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           1            1            0            0
 ./qiskit/test/mock/backends/bogota/defs_bogota.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           1            1            0            0
 ./qiskit/test/mock/backends/paris/conf_paris.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             1            1            0            0
 ./qiskit/test/mock/backends/belem/conf_belem.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             1            1            0            0
 ./qiskit/test/mock/backends/cambridge/props_cambridge.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    1            1            0            0
 ./qiskit/test/mock/backends/cambridge/props_cambridge_alt.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                1            1            0            0
 ./qiskit/test/mock/backends/armonk/props_armonk.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          1            1            0            0
 ./qiskit/test/mock/backends/armonk/conf_armonk.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           1            1            0            0
 ./qiskit/test/mock/backends/armonk/defs_armonk.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           1            1            0            0
 ./qiskit/test/mock/backends/casablanca/props_casablanca.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  1            1            0            0
 ./qiskit/test/mock/backends/boeblingen/props_boeblingen.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  1            1            0            0
 ./qiskit/test/mock/backends/poughkeepsie/props_poughkeepsie.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                              1            1            0            0
 ./qiskit/test/mock/backends/almaden/conf_almaden.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         1            1            0            0
 ./qiskit/test/mock/backends/mumbai/conf_mumbai.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           1            1            0            0
 ./qiskit/test/mock/backends/montreal/conf_montreal.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       1            1            0            0
 ./qiskit/test/mock/backends/athens/defs_athens.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           1            1            0            0
 ./qiskit/test/mock/backends/lima/defs_lima.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               1            1            0            0
 ./qiskit/test/mock/backends/belem/defs_belem.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             1            1            0            0
 ./qiskit/test/mock/backends/melbourne/conf_melbourne.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     1            1            0            0
 ./qiskit/test/mock/backends/jakarta/defs_jakarta.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         1            1            0            0
 ./qiskit/test/mock/backends/almaden/props_almaden.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        1            1            0            0
 ./qiskit/test/mock/backends/lagos/props_lagos.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            1            1            0            0
 ./qiskit/test/mock/backends/toronto/conf_toronto.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         1            1            0            0
 ./qiskit/test/mock/backends/valencia/conf_valencia.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       1            1            0            0
 ./qiskit/test/mock/backends/london/conf_london.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           1            1            0            0
 ./qiskit/test/mock/backends/rochester/props_rochester.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    1            1            0            0
 ./qiskit/test/mock/backends/sydney/props_sydney.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          1            1            0            0
 ./qiskit/test/mock/backends/manhattan/conf_manhattan.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     1            1            0            0
 ./qiskit/test/mock/backends/singapore/conf_singapore.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     1            1            0            0
 ./qiskit/test/mock/backends/rome/conf_rome.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               1            1            0            0
 ./qiskit/test/mock/backends/manila/conf_manila.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           1            1            0            0
 ./qiskit/test/mock/backends/mumbai/props_mumbai.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          1            1            0            0
 ./qiskit/test/mock/backends/manila/props_manila.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          1            1            0            0
 ./qiskit/test/mock/backends/guadalupe/conf_guadalupe.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     1            1            0            0
 ./qiskit/test/mock/backends/vigo/conf_vigo.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               1            1            0            0
 ./qiskit/test/mock/backends/paris/props_paris.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            1            1            0            0
 ./qiskit/test/mock/backends/santiago/props_santiago.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      1            1            0            0
 ./qiskit/test/mock/backends/guadalupe/props_guadalupe.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    1            1            0            0
 ./qiskit/test/mock/backends/melbourne/props_melbourne.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    1            1            0            0
 ./qiskit/test/mock/backends/boeblingen/defs_boeblingen.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   1            1            0            0
 ./qiskit/test/mock/backends/toronto/props_toronto.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        1            1            0            0
 ./qiskit/test/mock/backends/montreal/props_montreal.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      1            1            0            0
 ./qiskit/test/mock/backends/brooklyn/props_brooklyn.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      1            1            0            0
 ./qiskit/test/mock/backends/poughkeepsie/conf_poughkeepsie.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                               1            1            0            0
 ./qiskit/test/mock/backends/sydney/conf_sydney.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           1            1            0            0
 ./qiskit/test/mock/backends/singapore/props_singapore.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    1            1            0            0
 ./qiskit/test/mock/backends/tenerife/props_tenerife.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      1            1            0            0
 ./qiskit/test/mock/backends/vigo/props_vigo.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              1            1            0            0
 ./qiskit/test/mock/backends/lima/props_lima.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              1            1            0            0
 ./qiskit/test/mock/backends/essex/props_essex.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            1            1            0            0
 ./qiskit/test/mock/backends/yorktown/conf_yorktown.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       1            1            0            0
 ./qiskit/test/mock/backends/mumbai/defs_mumbai.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           1            1            0            0
 ./qiskit/test/mock/backends/paris/defs_paris.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             1            1            0            0
 ./qiskit/test/mock/backends/sydney/defs_sydney.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           1            1            0            0
 ./qiskit/test/mock/backends/manila/defs_manila.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           1            1            0            0
 ./qiskit/test/mock/backends/brooklyn/conf_brooklyn.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       1            1            0            0
 ./qiskit/test/mock/backends/casablanca/defs_casablanca.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   1            1            0            0
 ./qiskit/test/mock/backends/johannesburg/props_johannesburg.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                              1            1            0            0
 ./qiskit/test/mock/backends/montreal/defs_montreal.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       1            1            0            0
 ./qiskit/test/mock/backends/manhattan/defs_manhattan.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     1            1            0            0
 ./qiskit/test/mock/backends/brooklyn/defs_brooklyn.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       1            1            0            0
 ./qiskit/test/mock/backends/toronto/defs_toronto.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         1            1            0            0
 ./qiskit/test/mock/backends/poughkeepsie/defs_poughkeepsie.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                               1            1            0            0
 ./qiskit/test/mock/backends/lagos/defs_lagos.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             1            1            0            0
 ./test/ipynb/mpl/circuit/user_style.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    213          213            0            0
 ./qiskit/test/mock/backends/guadalupe/defs_guadalupe.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     1            1            0            0
 ./qiskit/test/mock/backends/manhattan/props_manhattan.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    1            1            0            0
 ./qiskit/test/mock/backends/johannesburg/defs_johannesburg.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                               1            1            0            0
 ./qiskit/test/mock/backends/singapore/defs_singapore.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     1            1            0            0
 ./qiskit/test/mock/backends/almaden/defs_almaden.json                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         1            1            0            0
========================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================
 Total                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          105         1121         1121            0            0
========================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================```